KifuSF/CreateDonationViewController.swift

```diff
@@ -0,0 +1,246 @@
+        configureGestures()
+        configureDelegates()
+
+        imageHelper.completionHandler = { [unowned self] (image) in
+            self.descriptorView.imageView.image = image
+            self.userSelectedAProfileImage = true
+        }
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        NotificationCenter.default.addObserver(self,selector: #selector(keyboardWillShow(_:)),
+                                               name: .UIKeyboardWillShow, object: nil)
+
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillHide(_:)),
+                                               name: .UIKeyboardWillHide, object: nil)
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        NotificationCenter.default.removeObserver(self, name: .UIKeyboardWillShow, object: nil)
+        NotificationCenter.default.removeObserver(self, name: .UIKeyboardWillHide, object: nil)
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        contentScrollView.updateBottomPadding(pickUpAddressButton.frame.height + 24)
+    }
+
+    //MARK: - Methods
+    @objc func keyboardWillShow(_ notification: Notification) {
+        guard let keyboardHeight = notification.getKeyboardHeight() else {
+            return assertionFailure("Could not retrieve keyboard height")
+        }
+
+        contentScrollView.updateBottomPadding(keyboardHeight + 20)
+    }
+
+    //TODO: this method gets called twice find out why
+    @objc private  func keyboardWillHide(_ notification: Notification) {
+        contentScrollView.updateBottomPadding(pickUpAddressButton.frame.height + 24)
+    }
+
+    @objc private func dismissVC() {
+        dismiss(animated: true)
+    }
+
+    @objc private func dismissKeyboard() {
+        view.endEditing(true)
+    }
+
+    @objc private func imageViewTapped() {
+        imageHelper.presentActionSheet(from: self)
+    }
+
+    @objc private func pickUpAddressButtonTapped() {
+        guard let _ = userSelectedAProfileImage,
+            let image = descriptorView.imageView.image,
+            let title = titleInputView.contentView.textField.text, !title.isEmpty,
+            let description = descriptionInputView.contentView.text, description != CreateDonationViewController.descriptionPlaceholder else {
+                return showErrorMessage("Please complete all the fields")
+        }
+
+        let locationPicker = LocationPickerViewController()
+
+        locationPicker.showCurrentLocationInitially = true
+        locationPicker.searchBarPlaceholder = "Choose Pickup location"
+        locationPicker.mapType = .standard
+
+        locationPicker.showCurrentLocationButton = true
+
+        locationPicker.completion = { location in
+            guard let location = location else {
+                return print("no location selected")
```

Comment:
```suggestion
                return showErrorMessage("Please select a pick-up location")
```

Since the user can open the location picker then click back without selecting a location

Reply:
Actually this completion handler doesn't get called if you don't select a location, it's more of a serious bug thingy

Reply:
I see, thanks for letting me know about the behavior of LocationPicker :)